extensions/src/main/java/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -491,6 +492,27 @@ public Response doGetFileOrDir( @PathParam ( "pathId" ) String pathId ) {
     }
   }
 
+  @GET
+  @Path( "/does-file-exists" )
```

Comment:
it might be redundant to include the word "file(s)" again in the url endpoint
The base class already has` "/repo/files/"`

so this new endpoint will be `"/repo/files/does-file-exists/pathId=XYZ"`
 `"/repo/files/exists/pathId=XYZ"` might be simplier